src/Security/Security.php

```diff
@@ -599,87 +747,172 @@ public function searchUsers(array $filters, array $options = [])
      * @param string $id Unique profile identifier
      * @param array $policies List of policies to apply to this profile
      * @param array $options Optional arguments
+     *
      * @return Profile
+     *
+     * @throws InvalidArgumentException
      */
     public function updateProfile($id, array $policies, array $options = [])
     {
-        $data = [
-            '_id' => $id,
-            'body' => [ 'policies' => $policies ]
-        ];
+        if (empty($id) || empty($policies)) {
+            throw new InvalidArgumentException('Kuzzle\Security::updateProfile: id and policies are required');
+        }
+
+        $options['httpParams'] = [':_id' => $id];
 
         $response = $this->kuzzle->query(
-            $this->buildQueryArgs('updateProfile'),
-            $data,
+            $this->kuzzle->buildQueryArgs('security', 'updateProfile'),
+            [
+                '_id' => $id,
+                'body' => json_encode([ 'policies' => $policies ])
+            ],
             $options
         );
 
         return new Profile($this, $id, $response['result']['_source'], $response['result']['_meta']);
     }
 
+    /**
+     * Performs an update on the internal profiles collection mapping.
+     *
+     * @param array $mapping New mapping
+     * @param array $options Optional arguments
+     *
+     * @return bool
+     *
+     * @throws InvalidArgumentException
+     */
+    public function updateProfileMapping(array $mapping = [], array $options = [])
```

Comment:
To be consistent with other SDKs, the first argument should be an associative array `$body` , which contains a `properties` key.

and return the whole `$response['result']`